runtime/src/tests.rs

```diff
@@ -100,3 +168,204 @@ fn pallet_poe_availability() {
+                    .0
+                    == offender_account
+            });
+            assert!(!found);
+
+            // AURA_AUTHOR_ID is considered offline
+            assert!(!ImOnline::is_online(AURA_AUTHOR_ID));
+
+            // Advance to next session
+            System::set_block_number(System::block_number() + 1);
+            Session::rotate_session();
+
+            // Check that the offline offence for the last session was received by pallet_offences
+            found = pallet_offences::ConcurrentReportsIndex::<Runtime>::get(
+                b"im-online:offlin",
+                session.encode(),
+            )
+            .into_iter()
+            .any(|offender| {
+                pallet_offences::Reports::<Runtime>::get(offender)
+                    .expect("Offence not found")
+                    .offender
+                    .0
+                    == offender_account
+            });
+            assert!(found);
+        });
+    }
+
+    pub const TEST_SLASH_FRACTION: Perbill = Perbill::one();
+    struct TestOffence {
+        offender_account: AccountId32,
+    }
+    impl offence::Offence<(AccountId32, Exposure<AccountId32, u128>)> for TestOffence {
+        const ID: offence::Kind = *b"testoffencenooop";
+        type TimeSlot = u128;
+
+        fn offenders(&self) -> Vec<(AccountId32, Exposure<AccountId32, u128>)> {
+            let exposure =
+                pallet_staking::EraInfo::<Runtime>::get_full_exposure(0, &self.offender_account);
+
+            vec![(self.offender_account.clone(), exposure)]
+        }
+        fn validator_set_count(&self) -> u32 {
+            NUM_VALIDATORS
+        }
+        fn time_slot(&self) -> Self::TimeSlot {
+            0
+        }
+        fn session_index(&self) -> SessionIndex {
+            0
+        }
+        fn slash_fraction(&self, _offenders_count: u32) -> Perbill {
+            TEST_SLASH_FRACTION
+        }
+    }
+
+    #[test]
+    fn offences_notifies_staking() {
+        new_test_ext().execute_with(|| {
+            let offender_account = sp_runtime::AccountId32::new(
+                testsfixtures::SAMPLE_USERS[AURA_AUTHOR_ID as usize]
+                    .raw_account
+                    .into(),
+            );
+
+            // Make sure that no slash events for offender_account is published
+            assert!(!System::events().contains(&EventRecord {
+                phase: Phase::Initialization,
+                event: RuntimeEvent::Staking(pallet_staking::Event::SlashReported {
+                    validator: offender_account.clone(),
+                    fraction: Perbill::one(),
+                    slash_era: 0,
+                }),
+                topics: vec![],
+            }));
+
+            // Make pallet_offences report an offence
+            let offence = TestOffence {
+                offender_account: offender_account.clone(),
+            };
+            assert!(Offences::report_offence(vec![], offence).is_ok());
```

Comment:
You can use `assert_ok!()` from `frame_support` here.

Reply:
Done